gui/src/types.ts

```diff
@@ -66,7 +68,10 @@ export const mnemonicSchema = z
       message:
         "Invalid number of words. Needs to be 12, 15, 18, 21 or 24 words long",
     }
-  );
+  )
+  .refine((data) => validateMnemonic(data, wordlist), {
+    message: "This word combination is not valid",
+  });
```

Comment:
hmm, this looks very much like what I would expect, but I was aiming to have this be done by the backend, instead of introducing extra dependencies on the UI

the reason being that the rust code is the source of truth, and the one that will ultimately parse the mnemonic. if there's some small inconsistency, it could lead to problems)

what I was thinking was adding a `iron_wallets::utils::validate_mnemonic` function, expose it as a command, and do:
```
refune(async (mnemonic) => await invoke("wallets_is_mnemonic_valid", { mnemonic })
```

I'll probably end up merging this as is and leaving that for future work, though